packages/jaeger-ui/src/model/ddg/DdgVertex.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { TVertex } from '@jaegertracing/plexus/lib/types';
+
+import { DdgEdge, PathElem } from './types';
+
+type TDdgVertex = {
+  egressEdges: Map<DdgVertex, DdgEdge>;
+  ingressEdges: Map<DdgVertex, DdgEdge>;
+  pathElems: Set<PathElem>;
+};
+
+export default class DdgVertex implements TVertex<TDdgVertex> {
```

Comment:
It works implicitly without `implements`, but I am wondering if `implements` adds some explicit clarity that this class will be used/usable where plexus takes vertices.
Based on [this article you sent me earlier](https://medium.com/@martin_hotell/interface-vs-type-alias-in-typescript-2-7-2a8f1777af4c) it seems as though implements works just as well for types and interfaces.

Reply:
In this case it seems okay because the `TDdgVertex` is not exported. But, if it were, it would likely cause some confusion.

It's also somewhat confusing in that `TDdgVertex` is a partial of `DdgVertex`. So, maybe the following would be more clear?

```js
type TDdgVertex = TVertex<{
  egressEdges: Map<DdgVertex, DdgEdge>;
  ingressEdges: Map<DdgVertex, DdgEdge>;
  pathElems: Set<PathElem>;
}>;

export default class DdgVertex implements TDdgVertex {
//...
}
```

Reply:
that is clearer.